File: nan21.dnet.module.bp/nan21.dnet.module.bp.presenter/src/main/java/net/nan21/dnet/module/bp/base/ds/model/CompanyLegalFormLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2010 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.bp.base.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.api.model.IModelWithClientId;
import net.nan21.dnet.core.api.model.IModelWithId;
import net.nan21.dnet.core.presenter.model.AbstractDsModel;

import net.nan21.dnet.module.bp.base.domain.entity.CompanyLegalForm;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = CompanyLegalForm.class, jpqlWhere = " e.active = true ", sort = { @SortField(field = CompanyLegalFormLovDs.fNAME) })
public class CompanyLegalFormLovDs extends AbstractDsModel<CompanyLegalForm>
        implements IModelWithId, IModelWithClientId {

    public static final String fID = "id";
    public static final String fCLIENTID = "clientId";
    public static final String fNAME = "name";
    public static final String fACTIVE = "active";
    public static final String fCOUNTRYID = "countryId";

    @DsField()
    private Long id;

    @DsField()
    private Long clientId;

    @DsField()
    private String name;

    @DsField()
    private Boolean active;

    @DsField(join = "left", path = "country.id")
    private Long countryId;

    public CompanyLegalFormLovDs() {
        super();
    }

    public CompanyLegalFormLovDs(CompanyLegalForm e) {
        super(e);
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Object id) {
        this.id = this._asLong_(id);

    }

    public Long getClientId() {
        return this.clientId;
    }

    public void setClientId(Long clientId) {
        this.clientId = clientId;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Boolean getActive() {
        return this.active;
    }

    public void setActive(Boolean active) {
        this.active = active;
    }

    public Long getCountryId() {
        return this.countryId;
    }

    public void setCountryId(Long countryId) {
        this.countryId = countryId;
    }

}

File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/webapp/src/ds/CompanyLegalFormLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.md.bp.ds.model.CompanyLegalFormLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"clientId", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"active", type:"boolean"},
		{name:"countryId", type:"int", useNull:true}
	]
});

Ext.define("net.nan21.dnet.module.md.bp.ds.model.CompanyLegalFormLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"countryId", type:"int", useNull:true},
		{name:"countryId_From",type:"int", useNull:true},
		{name:"countryId_To",type:"int", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.md.bp.ds.param.CompanyLegalFormLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 